### PR TITLE
Update junit to 4.13.1

### PR DESCRIPTION
**Pull request by lezzago**
Signed-off-by: Ashish Agrawal <ashisagr@amazon.com>

### Description
Update `junit` from `4.12` to `4.13.2`
 
### Issues Resolved
N/A
 
### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: buildSrc/version.properties [hunk 1]
```diff
@@ -28,7 +28,7 @@ joda              = 2.10.12
 bouncycastle=1.70
 # test dependencies
 randomizedrunner  = 2.7.1
-junit             = 4.12
+junit             = 4.13.1
```
Comment (reta) [inline buildSrc/version.properties:31]: Probably 4.13.2 would be better :-)

Comment (lezzago) [reply buildSrc/version.properties:31]: ahh good catch

**Comment by tlfeng** [comment]
In test report 1757:
```
* What went wrong:
Execution failed for task ':dependencyLicenses'.
> Missing SHA for junit-4.13.2.jar. Run "gradle updateSHAs" to create them
```

Hi @lezzago , please run `./gradlew updateSHAs` to update the checksum for the jar file. 😁

**Comment by lezzago** [comment]
> In test report 1757:
> 
> ```
> * What went wrong:
> Execution failed for task ':dependencyLicenses'.
> > Missing SHA for junit-4.13.2.jar. Run "gradle updateSHAs" to create them
> ```
> 
> Hi @lezzago , please run `./gradlew updateSHAs` to update the checksum for the jar file. 😁

Oh thanks, I will do that. Also is this in the developing guide? If not, might be worthwhile to add this in there for others in the future.

**Comment by tlfeng** [comment]
> Oh thanks, I will do that. Also is this in the developing guide? If not, might be worthwhile to add this in there for others in the future.

It's not explicitly told in the developing guide, but it's shown in the test report. 😄 Thanks for your opinion, maybe we'd better find a place to mention it.

**Comment by tlfeng** [comment]
The above test failure is a known issue https://github.com/opensearch-project/OpenSearch/issues/1806.
I will re-triggert the check: start gradle check.

**Review by dblock — APPROVED** [review]

**Pull request merged**